packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutThreshold.tsx

```diff
@@ -426,65 +447,100 @@ export const ChartDonutThreshold: React.FunctionComponent<ChartDonutThresholdPro
     const dynamicHeight = donutHeight - (theme.pie.height - dynamicTheme.pie.height);
     return Math.round((donutHeight - dynamicHeight) / 2);
   }
 
+  // Returns the horizontal shift for the donut utilization legend
+  const getLegendDx = (dynamicTheme: ChartThemeDefinition, position: string) => {
+    const dynamicWidth = donutWidth - (theme.pie.width - dynamicTheme.pie.width);
+    switch (position) {
+      case 'right':
+        return getDonutDx(dynamicTheme, legendPosition) + Math.round((donutWidth - dynamicWidth) / 2);
+      default:
+        return 0;
+    }
+  };
+
+  // Returns the vertical shift for the donut utilization legend and subtitle
+  const getLegendAndSubTitleDy = (dynamicTheme: ChartThemeDefinition, position: string) => {
+    const dynamicWidth = donutWidth - (theme.pie.width - dynamicTheme.pie.width);
+    switch (position) {
+      case 'bottom':
+        return getDonutDy(dynamicTheme) + Math.round((donutWidth - dynamicWidth) / 2);
+      default:
+        return getDonutDy(dynamicTheme);
+    }
+  }
+
+  // Returns the horizontal shift for the donut utilization subtitle
+  const getSubTitleDx = (dynamicTheme: ChartThemeDefinition, position: string) => {
+    const dynamicWidth = donutWidth - (theme.pie.width - dynamicTheme.pie.width);
+    switch (position) {
+      case 'right':
+        return getDonutDx(dynamicTheme, legendPosition) + Math.round((donutWidth - dynamicWidth) / 2);
+      default:
+        return Math.round((donutWidth - dynamicWidth) / 2);
+    }
+  };
+
   // Render dynamic utilization donut cart
   const renderChildren = () =>
     React.Children.toArray(children).map(child => {
-      if (child.props) {
-        const datum = getData([{ ...child.props.data }]);
-        const orientation = child.props.donutOrientation || donutOrientation;
-        const dynamicTheme =
-          child.props.theme ||
-          getDonutThresholdDynamicTheme(child.props.themeColor || themeColor,
-            child.props.themeVariant || themeVariant);
-        return React.cloneElement(child, {
-          donutDx: child.props.donutDx || getDynamicDonutDx(dynamicTheme, orientation),
-          donutDy: child.props.donutDy || getDynamicDonutDy(dynamicTheme),
-          donutHeight: child.props.donutHeight || donutHeight - (theme.pie.height - dynamicTheme.pie.height),
-          donutOrientation: orientation,
-          donutWidth: child.props.donutWidth || donutWidth - (theme.pie.width - dynamicTheme.pie.width),
-          endAngle: child.props.endAngle || 360 * (datum[0]._y ? datum[0]._y / 100 : 100),
-          height: child.props.height || height,
-          showStatic: child.props.showStatic || false,
-          standalone: false,
-          theme: dynamicTheme,
-          width: child.props.width || width
-        });
+      const props = child.props;
+      if (!props) {
+        return child;
       }
-      return child;
+      const datum = getData([{ ...props.data }]);
```

Comment:
No need for spread operator:
```suggestion
      const datum = getData([props.data]);
```